lib/src/embedded/isolate_dispatcher.dart

```diff
@@ -156,9 +159,14 @@ class IsolateDispatcher {
           _inactiveIsolates.add(isolate);
           resource.release();
           _channel.sink.add(packet);
-        case 2:
+        default:
           _channel.sink.add(packet);
-          exit(exitCode);
+          exitCode = category;
+          if (_gracefulShutdown) {
+            _channel.sink.close();
```

Comment:
Shouldn't this also send kill messages to the other isolates?

Reply:
That is handled in `onDone` hook for the `_channel`, so after sink is closed, `kill` is sent.